gremlin-go/driver/driverRemoteConnection.go

```diff
@@ -100,6 +128,39 @@ func (driver *DriverRemoteConnection) submitBytecode(bytecode *bytecode) (Result
 	return driver.client.submitBytecode(bytecode)
 }
 
+// isSessionBound returns True if a DriverRemoteConnection is a Session
+func (driver *DriverRemoteConnection) isSessionBound() bool {
+	return driver.client.Session != ""
+}
+
+// CreateSession generates a new Session. sessionId stores the optional UUID param. It can be used to create a Session with a specific UUID.
+func (driver *DriverRemoteConnection) CreateSession(sessionId ...string) (*DriverRemoteConnection, error) {
+	if len(sessionId) > 1 {
+		return nil, errors.New("more than one Session ID specified. Cannot create Session with multiple UUIDs")
+	} else if driver.isSessionBound() {
+		return nil, errors.New("connection is already bound to a Session - child sessions are not allowed")
+	}
+
+	driver.client.logHandler.logger.Log(Info, "creating Session based connection")
+	sid := uuid.New().String()
```

Comment:
This naming isn't the greatest and isn't necessary. You can just assign it directly in your func using an if statement.